Adjust bbox to view center

YanjiePa opened the pull request.

YanjiePa committed:
Try best to make dancer in the center of the view by adjusting X and Y locations.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

YanjiePa committed:
Smooth bbox in class motion.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

YanjiePa committed:
Refactor class EffectDecision and class VideoGeneration by
1. Move make_effect_point_list_from_desc_list() from class VideoGeneration to class EffectDecision
2. Adjust scale/loc_x/loc_y in EffectDecision.make_effect_point_list_from_desc_list()
3. Pass bbox info to class EffectDecision
4. Add user defined cfg to class EffectDecision

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

YanjiePa committed:
Calculate default loc_y based on the average center of all bboxes.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

Kotomi-Du commented on effect_decision.py:
```diff
@@ -2,8 +2,29 @@
+                        'adjust_loc_y': True,   # adjust the upper boundary of bbox within view
+                        'adjust_scale': False}  # adjust the previously decided scale so that the scaled bbox is within view
+        else:
+            self.cfg = cfg
```
why need this? what kind of config file it will be like?

Kotomi-Du commented on effect_decision.py:
```diff
@@ -139,14 +160,163 @@ def get_effect_list(self, area_data, audio_beats, au_instance, framerate, group_
+            key_scale = last_key_scale
+            for effect in effect_desc['effect']:
+                if effect['type'] == 'zoom':
+                    key_scale = effect['scale'] if effect['scale'] > 1 else default_scale
```
any case the scale <= 1?

Kotomi-Du commented on effect_decision.py:
```diff
@@ -139,14 +160,163 @@ def get_effect_list(self, area_data, audio_beats, au_instance, framerate, group_
+            # 'start_from': start_from,
+            # 'end_to': end_to}
+
+            # 1. According to effect_desc, get the scale at start_from, keyframe, and end_to.
```
If we pass the scale and other info directly rather than pass the effect_desc, maybe we can reduce code from line 197 to 207?

Kotomi-Du commented on effect_decision.py:
```diff
@@ -139,14 +160,163 @@ def get_effect_list(self, area_data, audio_beats, au_instance, framerate, group_
+        return effect_points
+
+    def adjust_scale(self, frame_idx, scale, strategy=0):
+        # adjust the previously decided scale so that the scaled bbox is within view
```
if this works, does we need to adjust loc_y?

Kotomi-Du reviewed (commented):
The comment is just some of my confusion. No need to answer. Let's discuss on the meeting tomorrow.

Kotomi-Du commented on main.py:
```diff
@@ -15,6 +15,8 @@ def main(video_path, output_dir, debug):
     m = Motion(video_path=video_path, debug=debug)
     area_data = m.area_data
+    bboxes = m.bboxes
+    infer_frame_size = m.infer_frame_size
```
This information can also be got from ffmpeg information directly

Kotomi-Du reviewed (commented):
No need to reply,  let's discuss on the meeting tomorrow

Kotomi-Du commented:
```
Def base_effect_decision():
   Return self.key_list, self. start_list, end_list, scale_list


Def get_effect_decision():
       Dic= {}
	Return dic

Def adjust_loc_x():
       For k,s,e in zip(key_list, start_list, end_list):

Def adjust _loc_y ():

```

YanjiePa committed:
Don't expose cfg used by class EffectDecision.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

YanjiePa committed:
Delete adjust_scale, i.e., don't adjust the previously decided scale.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

YanjiePa committed:
1. Refine adjust_loc_y.
2. Add 'default_scale' to cfg.
    (1) If true, keep using default_scale for start_from/end_to.
    (2) Otherwise, calculate the best scale for start_from/end_to.

Signed-off-by: Pan, Yanjie <yanjie.pan@intel.com>

Kotomi-Du commented:
It's nice to see the structure of current code is easy to understand. Just a little confused of the function name of [get_effect_point_list](https://github.com/Kotomi-Du/DanceU/blob/0e817e15bb68795178e8646da76016e8d0d3b6ee/effect_decision.py#L232)
